### PR TITLE
carma-platform truck inspection client - Truck VIN_Number is empty

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
Run the truck_inspection_client node and send MobilityRequest. Wait for 30 seconds, the vin_number does not exist error message send to system_alert. 

rosparam load <path to UniqueVehicleParams>/UniqueVehicleParams.ymal file, then send  a MobilityRequest again. The vin_number should be loaded, and sent to outgoing_mobility_operation topic.

## Description
vin_number is called from truck_inspection_client before it is loaded from UniqueVehicleParams.yml

## Related Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/992

## Motivation and Context
ACE testing

## How Has This Been Tested?
Local integration testing by running truck_inspection_client node

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by maefromm — APPROVED** [review]

File: truck_inspection_client/launch/truck_inspection_client.launch [hunk 1]
```diff
@@ -20,7 +20,7 @@
     <remap from="incoming_mobility_request" to="$(optenv CARMA_MSG_NS)/incoming_mobility_request"/>
     <remap from="guidance/state" to="$(optenv CARMA_GUIDE_NS)/state"/>
 
-    <node name="truck_inspection_client" pkg="truck_inspection_client" type="truck_inspection_client">
+    <node name="truck_inspection_client" pkg="truck_inspection_client" type="truck_inspection_client"  output="screen">
```
Comment (msmcconnell) [inline truck_inspection_client/launch/truck_inspection_client.launch:23]: Don't change this to output=screen. Log files are easier to parse through when errors occur.

File: truck_inspection_client/src/truck_inspection_client.cpp [hunk 2]
```diff
@@ -36,6 +36,7 @@ namespace truck_inspection_client
         pnh_->getParam("permit_required", permit_required_);
         pnh_->getParam("pre_trip_ads_health_check", pre_trip_ads_health_check_);
         mo_pub_ = nh_->advertise<cav_msgs::MobilityOperation>("outgoing_mobility_operation", 5);
+        vin_system_alert_pub_ = nh_->advertise<cav_msgs::SystemAlert>("system_alert", 10);
```
Comment (msmcconnell) [inline truck_inspection_client/src/truck_inspection_client.cpp:39]: It is not necessary to create your own system_alert publisher. CARMANodeHandle does that by default.
You can just call
nh_->publishSystemAlert(alert_msg);

File: truck_inspection_client/src/truck_inspection_client.cpp [hunk 3]
```diff
@@ -79,6 +80,30 @@ namespace truck_inspection_client
...
+                //sleep for 0.1 second
+                boost::this_thread::sleep_for(boost::chrono::milliseconds(100));
+                vin_retrive_count++;
+
+                if(pnh_->getParam("/vin_number", vin_number_))
+                {
+                    ROS_INFO("retrieved vin_number1:  %s",vin_number_.c_str());
+                    break; 
+                }
+                ROS_WARN("retrieving vin_number:  %s",vin_number_.c_str());
```
Comment (msmcconnell) [inline truck_inspection_client/src/truck_inspection_client.cpp:96]: Lets make this a debug log. The info and error logs are really sufficient to describe what is happening.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: truck_inspection_client/src/truck_inspection_client.cpp [hunk 4]
```diff
@@ -79,6 +80,30 @@ namespace truck_inspection_client
     void TruckInspectionClient::requestCallback(const cav_msgs::MobilityRequestConstPtr& msg)
     {
         if(msg->strategy == this->INSPECTION_STRATEGY) {
+            std::uint16_t vin_retrive_count = 0;
+            //Checking: vin number does not exist yet in global parameter server
+            while((!pnh_->getParam("/vin_number", vin_number_)) && vin_retrive_count < MAX_RETRIEVE_VIN_COUNT)
```
Comment (msmcconnell) [inline truck_inspection_client/src/truck_inspection_client.cpp:85]: This loop should be located in the initialize() method. Not hear. If the parameter is not loaded by this point it never will be.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: truck_inspection_client/launch/truck_inspection_client.launch (hunk 1 above)
Comment (dan-du-car) [reply truck_inspection_client/launch/truck_inspection_client.launch]: Addressed: https://github.com/usdot-fhwa-stol/carma-platform/pull/1020/commits/56dd79335ca2255d024912c27d7e13719a88414d

File: truck_inspection_client/src/truck_inspection_client.cpp (hunk 2 above)
Comment (dan-du-car) [reply truck_inspection_client/src/truck_inspection_client.cpp]: Addressed: https://github.com/usdot-fhwa-stol/carma-platform/pull/1020/commits/56dd79335ca2255d024912c27d7e13719a88414d

File: truck_inspection_client/src/truck_inspection_client.cpp (hunk 4 above)
Comment (dan-du-car) [reply truck_inspection_client/src/truck_inspection_client.cpp]: Addressed: https://github.com/usdot-fhwa-stol/carma-platform/pull/1020/commits/56dd79335ca2255d024912c27d7e13719a88414d

**Review by msmcconnell — APPROVED** [review]

**Review by maefromm — APPROVED** [review]

**Pull request merged**